File: 2490_circular_sentence.java
class Solution {
    public boolean isCircularSentence(String s) {
        return s.charAt(0) == s.charAt(s.length() - 1) && IntStream.range(1, s.length() - 1).allMatch(i -> s.charAt(i) != ' ' || s.charAt(i - 1) == s.charAt(i + 1));
    }
}

File: 2490_circular_sentence.cpp
class Solution {
public:
    bool isCircularSentence(string& s) {
        return s.front() == s.back() && all_of(s.begin(), s.end(), [&, i=1](char) mutable {
            return s[i++] != ' ' || s[i-2] == s[i];
        });
    }
};
